flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -130,11 +152,17 @@ public AsyncExecutionController(
                                         },
                                         "AEC-buffer-timeout"));
 
+        if (parallelMode == ParallelMode.SERIAL_BETWEEN_EPOCH) {
```

Comment:
Whether two epochs can be executed in parallel depends on the semantics of the non-record in between which means it is not a unified configuration that can be set on the whole operator.

Reply:
Yes, thanks for the suggestion, I put `ParallelMode` as one parameter of `onNonRecord()`, then we can choose different modes for different epochs.